sync/src/relayer/mod.rs

```diff
@@ -836,12 +836,24 @@ impl CKBProtocolHandler for Relayer {
             }
             ASK_FOR_TXS_TOKEN => self.ask_for_txs(nc.as_ref()),
             TX_HASHES_TOKEN => self.send_bulk_of_tx_hashes(nc.as_ref()),
-            SEARCH_ORPHAN_POOL_TOKEN => tokio::task::block_in_place(|| {
-                self.shared.try_search_orphan_pool(
-                    &self.chain,
-                    &self.shared.active_chain().tip_header().hash(),
-                )
-            }),
+            SEARCH_ORPHAN_POOL_TOKEN => {
+                if !self.shared.state().orphan_pool().is_empty() {
+                    tokio::task::block_in_place(|| {
+                        self.shared.try_search_orphan_pool(
+                            &self.chain,
+                            &self.shared.active_chain().tip_header().hash(),
+                        );
+                        for hash in self.shared.state().orphan_pool().exist_parents() {
+                            if self.shared.state().orphan_pool().is_empty() {
+                                break;
+                            }
+                            if self.shared.is_stored(&hash) {
+                                self.shared.try_search_orphan_pool(&self.chain, &hash);
+                            }
+                        }
+                    })
+                }
+            }
```

Comment:
suggest to extract these lines and move it to `shared.try_search_orphan_pool` fn, and this fn should be refactor to:
```
pub fn try_search_orphan_pool(&self, chain: &ChainController) {

}
```

Reply:
`insert_new_block` also use this api，It needs to keep the current API